### PR TITLE
refactored dashboard ExportSearchResultsPage

### PR DESCRIPTION
**Pull request by lbownik**
refactored dashboard ExportSearchResultsPage

### CODE REVIEW

File: dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/JpaRepository.java [hunk 1]
```diff
@@ -15,6 +15,8 @@
 
 import static java.util.Optional.ofNullable;
 
+import java.util.Collection;
```
Comment (diasf) [inline dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/JpaRepository.java]: Unused import.

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DataverseSession.java [hunk 2]
```diff
@@ -94,6 +94,10 @@ public String getLocaleTitle() {
     public int getFilesPerPage() {
         return filesPerPage;
     }
+    
+    public boolean canEditDashboard() {
```
Comment (diasf) [inline dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DataverseSession.java]: Should be moved to the LOGIC section below.

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java [hunk 3]
```diff
@@ -25,24 +25,20 @@
...
-    private DashboardExportSearchResultsPage page;
-
     @Mock
     private DataverseSession session;
     @Mock
     private SystemConfig systemConfig;
     @Mock
     private DatasetFieldTypeRepository datasetFiledTypeRepo;
     @Mock
-    private PermissionsWrapper permissionsWrapper;
```
Comment (diasf) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java:39]: You still need a `NavigationWrapper` mock.

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java [hunk 4]
```diff
@@ -55,28 +51,29 @@ public void setUp() {
         this.type2.setId(2L);
         this.type2.setTitle("abc");
 
-        when(this.datasetFiledTypeRepo.findAll())
-                .thenReturn(asList(this.type1, this.type2));
+        when(this.datasetFiledTypeRepo.findAll()).thenReturn(types);
+
+        this.page = new DashboardExportSearchResultsPage(this.session, null,
```
Comment (diasf) [inline dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java]: Why did you remove `@InjectMocks`?

File: dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/JpaRepository.java (hunk 1 above)
Comment (lbownik) [reply dataverse-persistence/src/main/java/edu/harvard/iq/dataverse/persistence/JpaRepository.java]: fixed

File: dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DataverseSession.java (hunk 2 above)
Comment (lbownik) [reply dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DataverseSession.java]: done

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java (hunk 3 above)
Comment (lbownik) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java:39]: I dont need it since I don't test DashboardExportSearchResultsPage.verifyAccess() method, and I don't intend to for now.

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java (hunk 4 above)
Comment (lbownik) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java]: the constructor  of DashboardExportSearchResultsPage invokes datasetFiledTypeRepo.findAll().

i need to call when(this.datasetFiledTypeRepo.findAll()).thenReturn(types); BEFORE instantiating DashboardExportSearchResultsPage 
otherwise datasetFiledTypeRepo.findAll() returns empty list

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java (hunk 3 above)
Comment (diasf) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java:39]: But you're purposefully breaking the constructor interface by providing null and you're still creating a `SystemConfig` mock even though it doesn't seem necessary. I mean, it's not very consistent. Tests are in a way much like production code.

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java (hunk 4 above)
Comment (diasf) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java]: I guess the problem stems from using logic inside the constructor,  which isn't always the best choice (specifically in a dependency injection context). For instance, you could have extracted the logic to a `@PostCostruct` annotated method and call that method here to make the intend (logic usage) more obvious.

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java (hunk 3 above)
Comment (lbownik) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java:39]: fixed

File: dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java (hunk 4 above)
Comment (lbownik) [reply dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dashboard/DashboardExportSearchResultsPageTest.java]: fixed

**Review by diasf — APPROVED** [review]
